### PR TITLE
fix: use `set -euo pipefail` for pacscript functions

### LINKED ISSUE #537: Pacscripts should run with 'set -e'
### Is your feature related to a problem? Please describe.

Currently, all errors in Pacscripts are ignored unless explicitly caught with a `return` statement. This default is almost never the correct behavior, and has the potential to mask packaging errors, such as files being installed into nonexistent directories, trying to install files that do not exist, or any compilation errors. This makes it impractical to test Pacscript upgrades in a noninteractive way, since the output of Pacstall must be manually checked to see if any errors occurred that were logged but did not abort execution.

### A clear and concise description of what you want to happen.

By default, all Pacscript hooks should execute in a `set -e` environment. In fact, I would advocate for `set -euo pipefail`, since most programmers generally expect referring to a nonexistent variable to cause an error, and since the non-pipefail behavior (i.e. ignore all errors that may occur in commands on the left-hand side of a pipeline) is a repeated surprise to programmers new to Bash.

This will obviously be a breaking change and will require a number of Pacscripts to be updated, but I think it is an important enough change to justify the inconvenience. To ease the migration, we could make the new default an opt-in to begin with, then migrate over core Pacscripts one at a time, and finally flip the default as a major version bump.

### Describe alternatives you've considered

The main alternative to `set -e` is suffixing every line of every shell script with `|| return`. Some people advocate this approach since it is more explicit, but I tend to think it is at odds with the goal of Pacscript to deduplicate command packaging code and "do the right thing" by default.

### Additional context

`makepkg` uses `set -e` to execute hooks.

### PR DESCRIPTION
**Pull request by D-Brox**
## Purpose

Errors in pacscript functions are currently ignored unless explicitly caught with a return statement.

## Approach

Put the pacscript functions inside subshells with `set -euo pipefail`.

## Addendum

Closes #537

## Checklist

- [x] I confirm that I have read the [contributing guidelines](https://github.com/pacstall/pacstall/blob/develop/CONTRIBUTING.md), and this pull request is abiding by all the clauses stated in the guideline.

### CODE REVIEW

File: misc/scripts/install-local.sh [hunk 1]
```diff
@@ -610,7 +612,8 @@ fi
 trap - SIGINT
 
 fancy_message info "Installing"
-if ! install; then
+cd $(cat /tmp/pacstall-curdir)
```
Comment (Henryws) [inline misc/scripts/install-local.sh:615]: This can just be `cd < /tmp/pacstall-curdir`

**Review by Henryws — CHANGES_REQUESTED** [review]
Fix Codacy warnings.

**Review by Henryws — CHANGES_REQUESTED** [review]
```bash
henry@twilight ~ % ››› pacstall -I neofetch
Do you want to view/edit the pacscript [y/N] y
[+] INFO: Sourcing pacscript
[+] INFO: Creating dummy package
[+] INFO: Installing dependencies
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'neofetch' instead of '/tmp/pacstall/neofetch-pacstall.deb'
The following packages will be upgraded:
  neofetch
1 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/1276 B of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 /tmp/pacstall/neofetch-pacstall.deb neofetch all 7.1.0-1 [1276 B]
apt-listchanges: Reading changelogs...
(Reading database ... 533254 files and directories currently installed.)
Preparing to unpack .../pacstall/neofetch-pacstall.deb ...
Unpacking neofetch (7.1.0-1) over (7.1.0-1) ...
Setting up neofetch (7.1.0-1) ...
[+] INFO: Retrieving packages
Initializing download: https://github.com/dylanaraps/neofetch/archive/7.1.0.zip
File size: unavailable
Opening output file 7.1.0.zip
Server unsupported, starting from scratch with one connection.
Starting download

Connection 0 finished

Downloaded 94.9971 Kilobyte(s) in 0 second(s). (269.24 KB/s)
[+] INFO: Preparing
potato
neofetch.pacscript: line 18: eco: command not found
never gonna give you up
[+] INFO: Building
[+] INFO: Installing
[+] INFO: Symlinking files
[+] INFO: Storing pacscript
[+] INFO: Cleaning up
```
Continues after `eco` is not found

File: misc/scripts/install-local.sh [hunk 2]
```diff
@@ -247,7 +248,7 @@ if [[ -z $PACSTALL_REMOVE ]] && [[ -z $PACSTALL_INSTALL ]]; then
 	rm -rf '"$name"' 2> /dev/null
 	hash -r
 	if declare -F removescript >/dev/null ; then
-		removescript
+		(set -euo pipefail; removescript)
```
Comment (Henryws) [inline misc/scripts/install-local.sh:251]: Change the syntax to the `bash -ce "removescript"`, and make sure to export the function beforehand.

**Review by Henryws — CHANGES_REQUESTED** [review]

**Review by Henryws — CHANGES_REQUESTED** [review]
```bash
henry@twilight ~ % ››› pacstall -I treefetch-bin
Do you want to view/edit the pacscript [y/N] y
[+] INFO: Sourcing pacscript
[+] INFO: Creating dummy package
[+] INFO: Installing dependencies
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'treefetch-bin' instead of '/tmp/pacstall/treefetch-bin-pacstall.deb'
The following packages were automatically installed and are no longer required:
  catimg libcaca-dev libsixel-dev libslang2-dev python3-ldb python3-talloc
Use 'sudo apt autoremove' to remove them.
The following NEW packages will be installed:
  treefetch-bin
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/1436 B of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 /tmp/pacstall/treefetch-bin-pacstall.deb treefetch-bin all 2.0.0-1 [1436 B]
Selecting previously unselected package treefetch-bin.
(Reading database ... 533529 files and directories currently installed.)
Preparing to unpack .../treefetch-bin-pacstall.deb ...
Unpacking treefetch-bin (2.0.0-1) ...
Setting up treefetch-bin (2.0.0-1) ...
[+] INFO: Retrieving packages
Initializing download: https://github.com/angelofallars/treefetch/releases/download/v2.0.0/treefetch
File size: 5.14285 Megabyte(s) (5392672 bytes)
Opening output file treefetch
Starting download

Connection 0 finished
Connection 2 finished
Connection 3 finished
[100%] [..............................................................] [  13.1MB/s] [00:00]

Downloaded 5.14285 Megabyte(s) in 0 second(s). (13368.37 KB/s)
[*] WARNING: Could not enter into the downloaded archive
[+] INFO: Preparing
[+] INFO: Building
[+] INFO: Installing
install: target '/usr/src/pacstall/treefetch-bin/usr/bin/' is not a directory: No such file or directory
[!] ERROR: Could not install treefetch-bin properly
[+] INFO: Cleaning up
```

breaks `treefetch-bin`'s `install` function. Most likely because the inside of the function contains the `install` command

File: misc/scripts/remove.sh [hunk 3]
```diff
@@ -78,12 +86,16 @@ case "$url" in
...
-			if ! removescript; then
+			export -f ask fancy_message removescript
+			bash -ce "/var/cache/pacstall/${PACKAGE}/${_version}/${PACKAGE}.pacscript; removescript" || {
 				error_log 2 "removescript $PACKAGE"
-				fancy_message "Could not run removescript properly"
+				fancy_message error "Could not run removescript properly"
 				exit 1
-			fi
+			}
+			trap -
```
Comment (Henryws) [inline misc/scripts/remove.sh:97]: Doesn't this not work?

**Review by Henryws — CHANGES_REQUESTED** [review]

File: misc/scripts/install-local.sh [hunk 4]
```diff
@@ -683,14 +680,18 @@ fi
...
 		error_log 5 "postinst hook"
 		fancy_message error "Could not run postinst hook successfully"
 		sudo dpkg -r "$name" > /dev/null
 		fancy_message info "Cleaning up"
 		cleanup
 		exit 1
-	fi
+	}
+	trap -
+	trap - SIGINT
```
Comment (Henryws) [inline misc/scripts/install-local.sh:694]: 🤨

**Review by Henryws — CHANGES_REQUESTED** [review]

File: misc/scripts/install-local.sh [hunk 5]
```diff
@@ -38,6 +38,7 @@ function cleanup() {
 		sudo rm -rf "${SRCDIR:?}"/*
 		sudo rm -rf /tmp/pacstall/*
 	fi
+	sudo rm -f /tmp/pacstall-func
```
Comment (Henryws) [inline misc/scripts/install-local.sh:41]: This shouldn't be using sudo

**Review by Henryws — CHANGES_REQUESTED** [review]

File: misc/scripts/install-local.sh (hunk 5 above)
Comment (D-Brox) [reply misc/scripts/install-local.sh:41]: Actually, this shouldn't be here anymore after my last refactor

Comment (D-Brox) [reply misc/scripts/install-local.sh]: Nvm, I mistaken it with another file

**Review by Henryws — APPROVED** [review]

**Review by Henryws — APPROVED** [review]

**Review by wizard-28 — APPROVED** [review]
Reinstalled all currently installed applications. Works, but couldn't test all applications in the repository.

**Pull request merged**